File: spoj/classicals/AMR11E.py

def prime_factors(n):
	if not n%2:
		f = [2]
		while not n%2:
			n /= 2
	else:
		f=[]
	i=3
	while i*i <= n:
		if not n%i:
			f.append(i)
			while not n%i:
				n /= i
		i += 2
	if n!=1:f.append(n)
	return f

luckys=[]
for i in xrange(30,3000):
	if len(prime_factors(i)) >= 3:
		luckys.append(i)
		if len(luckys) >= 1000:
			break

for _ in xrange(input()):
	i = input()
	i -= 1
	print luckys[i]

File: spoj/classicals/AMR11E.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> prime_factors(int n){
	vector<int> f;
	if (n%2 == 0){
		f.push_back(2);
		while (n%2 == 0) n/= 2;
	}
	int i = 3;
	while (i*i <= n){
		if (n % i == 0){
			f.push_back(i);
			while (n%i == 0) n /= i;
		}
		i += 2;
	}
	if (n != 1) f.push_back(n);
	return f;
}

vector<int> luckys;

int main(void){

	for (int i = 30; i <= 30000; i += 1){
		vector<int> f = prime_factors(i);
		if (f.size() >= 3){
			luckys.push_back(i);
		}
		if (luckys.size() >= 1000) break;
	}
	int t,n;
	cin >> t;
	while (t--){
		cin >> n;
		n -= 1;
		cout << luckys[n] << endl;
	}

}


